ctru-rs/src/services/ndsp/wave.rs

```diff
@@ -0,0 +1,138 @@
+use ctru_sys::ndspChnWaveBufClear;
+
+use super::AudioFormat;
+use crate::linear::LinearAllocator;
+
+/// Base struct to represent audio wave data. This requires audio format information.
+pub struct WaveBuffer {
+    /// Buffer data. This data must be allocated on the LINEAR memory.
+    data: Box<[u8], LinearAllocator>,
```

Comment:
It's great that we're able to specify where this memory needs to be allocated in the type system!

Reply:
It is! Having RAM sectors we can freely allocate to is very nice, and this also means we can do static memory checks without any runtime cost.